contracts/FigmentEth2Depositor.t.sol

```diff
@@ -2,14 +2,954 @@
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
 
 contract FigmentEth2DepositorTest is Test {
-  FigmentEth2Depositor depositor;
+    FigmentEth2Depositor figmentDepositor;
+    MockDepositContract mockDepositContract;
 
-  function setUp() public {
-    // Mock deposit contract address for testing
-    address mockDepositContract = 0x00000000219ab540356cBB839Cbe05303d7705Fa;
-    depositor = new FigmentEth2Depositor(mockDepositContract);
-  }
+    function setUp() public {
+        // Deploy mock deposit contract for testing
+        mockDepositContract = new MockDepositContract();
+        figmentDepositor = new FigmentEth2Depositor(address(mockDepositContract));
+    }
+
+    // ============ Constructor Tests ============
+
+    function test_Constructor_Success() public {
+        address ethDepositContract = address(0x123);
+        FigmentEth2Depositor newFigmentEth2Depositor = new FigmentEth2Depositor(ethDepositContract);
+
+        assertEq(address(newFigmentEth2Depositor.depositContract()), ethDepositContract);
+        assertEq(newFigmentEth2Depositor.owner(), address(this));
+    }
+
+    function test_Constructor_ZeroAddress_Reverts() public {
+        vm.expectRevert(FigmentEth2Depositor.ZeroAddress.selector);
+        new FigmentEth2Depositor(address(0));
+    }
+
+    // ============ Receive Function Tests ============
+
+    function test_Receive_DirectEthTransfer_Reverts() public {
+        // Capture pre-transaction balances
+        uint256 preSenderBalance = address(this).balance;
+        uint256 preContractBalance = address(figmentDepositor).balance;
+        uint256 preMockBalance = address(mockDepositContract).balance;
+
+        vm.expectRevert(FigmentEth2Depositor.DirectEthTransferNotAllowed.selector);
+        address(figmentDepositor).call{value: 1 ether}("");
+
+        // Capture post-transaction balances
+        uint256 postSenderBalance = address(this).balance;
+        uint256 postContractBalance = address(figmentDepositor).balance;
+        uint256 postMockBalance = address(mockDepositContract).balance;
+
+        // Verify balances after revert
+        assertEq(postSenderBalance, preSenderBalance, "Sender balance should be unchanged after revert");
+        assertEq(postContractBalance, preContractBalance, "Contract balance should remain zero after revert");
+        assertEq(postMockBalance, preMockBalance, "Mock contract balance should be unchanged after revert");
+        assertEq(postContractBalance, 0, "Contract balance should be zero");
+    }
+
+    function test_Receive_DirectEthTransfer_WithData_Reverts() public {
+        // Capture pre-transaction balances
+        uint256 preSenderBalance = address(this).balance;
+        uint256 preContractBalance = address(figmentDepositor).balance;
+        uint256 preMockBalance = address(mockDepositContract).balance;
+
+        vm.expectRevert();
+        Address.functionCallWithValue(address(figmentDepositor), hex"1234", 1 ether);
```

Comment:
@mkornatz 
I removed the fallback call and also realized the vm.expectRevert has some bugs in verification of the error message when it gets into internal calls like this. In our case we can verify the the call reverts with the EVM revert and I think that should be enough without having to add another fallback call.